src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -576,4 +620,56 @@ object RuleUtils {
     assert(shuffleInjected)
     shuffled
   }
+
+  private def transformProject(project: Project, index: IndexLogEntry): Project = {
```

Comment:
Could you add some comment for these functions to help to understand?
(please do the same for other functions except for too simple & explicit one) Thanks!